packages/cli-kit/src/private/node/api/graphql.ts

```diff
@@ -17,20 +17,17 @@ export function graphqlRequest<T>(
   variables?: Variables,
   handleErrors = true,
 ): Promise<T> {
-  const action = async () => {
+  return (async () => {
     const headers = buildHeaders(token)
     debugLogRequestInfo(api, query, variables, headers)
     const clientOptions = {agent: await httpsAgent(), headers}
     const client = new GraphQLClient(url, clientOptions)
-    const response = await debugLogResponseInfo({request: client.rawRequest<T>(query as string, variables), url})
+    const response = await debugLogResponseInfo(
+      {request: client.rawRequest<T>(query as string, variables), url},
+      handleErrors ? errorHandler(api) : undefined,
+    )
     return response.data
-  }
-
-  if (handleErrors) {
-    return handlingErrors(api, action)
-  } else {
-    return action()
-  }
+  })()
```

Comment:
wait, why do we need the self-invoking function here?

Reply:
we have the change the function to be async, but it's returning a promise already, so I don't see why not :)